src/DDTrace/Util/HexConversion.php

```diff
@@ -6,6 +6,29 @@ final class HexConversion
 {
     public static function idToHex($id)
     {
-        return str_pad(dechex($id), 16, "0", STR_PAD_LEFT);
+        $hex = dd_trace_dec_hex((string) $id);
+        return str_pad($hex, 16, "0", STR_PAD_LEFT);
+    }
+
+    public static function hexToInt($hex)
+    {
+        if (strlen($hex) > 16) {
+            throw new \InvalidArgumentException(
+                '\HexConversion::hexToInt() arg must be hexstring with <= 16 characters'
+            );
+        }
+
+        if (strlen($hex) == 16 && strpos("89abcdef", $hex[0]) !== false) {
+            // over PHP_INT_MAX, so needs to be shifted right
+            // per original dd_trace_generate_id implementation.
+            $split = str_split($hex);
```

Comment:
My faded memories of PHP tell me that this may be quote slow since it creates and array of strings each one character length. Is there any benefit in doing this instead of just iterating over the string by index (i.e. `for ($i = 0; $i < strlen($hex); $i++){ $char=$hex[i]; ...`)?
I may be wrong though, so please correct me.

Reply:
Thanks, updated.